kratos/geometries/tetrahedra_3d_4.h

```diff
@@ -1410,6 +1418,70 @@ template<class TPointType> class Tetrahedra3D4 : public Geometry<TPointType>
                 rResult[i] = DN_DX;
     }
 
+    /**
+     * returns the second order derivatives of all shape functions
+     * in given arbitrary points
+     * @param rResult a third order tensor which contains the second derivatives
+     * @param rPoint the given point the second order derivatives are calculated in
+     */
+    ShapeFunctionsSecondDerivativesType& ShapeFunctionsSecondDerivatives( ShapeFunctionsSecondDerivativesType& rResult, const CoordinatesArrayType& rPoint ) const override
+    {
+        if ( rResult.size() != this->PointsNumber() )
+        {
+            // KLUDGE: While there is a bug in ublas vector resize, I have to put this beside resizing!!
+            ShapeFunctionsGradientsType temp( this->PointsNumber() );
+            rResult.swap( temp );
+        }
+
+        for ( unsigned int i = 0; i < this->PointsNumber(); i++ )
+        {
+            rResult[i].resize( 3, 3, false );
+            noalias( rResult[i] ) = ZeroMatrix( 3, 3 );
+        }
+
+        rResult[0]( 0, 0 ) = 0.0;
+        rResult[0]( 0, 1 ) = 0.0;
+        rResult[0]( 0, 2 ) = 0.0;
+        rResult[0]( 1, 0 ) = 0.0;
```

Comment:
Aren't they already all zeros? in the `for `above thay have been set to` ZeroMAtrix`

Reply:
Indeed

Reply:
Why not following @loumalouomega 's and @AlejandroCornejo 's suggestion in here? It would be much less verbose.